File: trunk/CodeModel/Generic/IProject.cs
/*
Copyright (c) 2008 Joa Ebert

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
THE SOFTWARE.
*/

using System;
using System.Collections.Generic;
using System.Text;

namespace AS3V.CodeModel.Generic
{
    public interface IProject
    {
        int RootCount { get; }
        IPackageRoot RootAt(int index);

        IEnumerable<IPackageRoot> RootEnumerator { get; }
        
        string Name { get; }
        string DirectoryPath { get; }

        bool HasCompilationUnit(string unitPath);
        ICompilationUnit CompilationUnit(string unitPath);

        bool HasPackage(string packagePath);
        IPackage Package(string packagePath);

        void Build(string projectPath, List<string> packageRoots);
        void Parse();
    }
}

File: trunk/java/src/com/joa_ebert/as3v/projectModel/IProject.java
package com.joa_ebert.as3v.projectModel;

import java.io.IOException;
import java.util.ArrayList;
import java.util.Iterator;

public interface IProject {

	int rootCount();

	IPackageRoot rootAt(int index);

	Iterator<IPackageRoot> rootIterator();

	
	/**
	 * Returns the name of the current project.
	 * 
	 * @return The name of the current project.
	 */
	String name();

	String fileSystemPath();

	boolean hasCompilationUnit(String unitPath);

	ArrayList<ICompilationUnit> compilationUnit(String unitPath);

	boolean hasPackage(String packagePath);
	
	ArrayList<IPackage> package$(String packagePath);
	
	void build(String fileSystemPath, ArrayList<String> packageRoots) throws IOException;
	
	void validate();
}
